### PR TITLE
Fixes crash + adds reminder + config change: infinite resources false

### PR DESCRIPTION
**Pull request by Raycoms**
Fixes the current gameCrash and sets the infinite resources of the builder to false in the config.

### CODE REVIEW

**Comment by Raycoms** [comment]
:shipit: please

**Comment by cltnschlosser** [comment]
:shipit:

File: src/main/java/com/minecolonies/entity/ai/basic/AbstractEntityAIStructure.java [hunk 1]
```diff
@@ -111,9 +111,10 @@ protected AbstractEntityAIStructure(@NotNull final J job)
      */
     private Supplier<AIState> generateSchematicIterator(Function<Structure.SchematicBlock, Boolean> evaluationFunction, AIState nextState)
     {
-        // do not replace with method reference, this one stays the same on changing reference for currentStructure
-        Supplier<Structure.SchematicBlock> getCurrentBlock = currentStructure::getCurrentBlock;
-        Supplier<Structure.Result>         advanceBlock    = currentStructure::advanceBlock;
+        //do not replace with method reference, this one stays the same on changing reference for currentStructure
+        //URGENT: DO NOT REPLACE FOR ANY MEANS THIS WILL CRASH THE GAME.
+        Supplier<Structure.SchematicBlock> getCurrentBlock = () -> currentStructure.getCurrentBlock();
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/entity/ai/basic/AbstractEntityAIStructure.java]: MINOR Replace this lambda with a method reference. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1612)

File: src/main/java/com/minecolonies/entity/ai/basic/AbstractEntityAIStructure.java [hunk 2]
```diff
@@ -111,9 +111,10 @@ protected AbstractEntityAIStructure(@NotNull final J job)
...
+        Supplier<Structure.Result>         advanceBlock    = () -> currentStructure.advanceBlock();
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/entity/ai/basic/AbstractEntityAIStructure.java]: MINOR Replace this lambda with a method reference. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1612)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 2 issues
* MINOR 2 minor

Watch the comments in this conversation to review them.

**Comment by Kostronor** [comment]
LGTM damn it sorry!

**Pull request merged**